src/core/Form/TextInput/TextInput.baseStyles.tsx

```diff
@@ -76,12 +80,18 @@ export const baseStyles = withSuomifiTheme(
 
     &.fi-text-input--error {
       & .fi-text-input_input {
-        border-color: ${theme.colors.alertBase};
+        border: 2px solid ${theme.colors.alertBase};
+        padding-left: ${math(`${theme.spacing.insetL} - 1`)};
+        padding-top: 7px;
+        padding-bottom: 7px;
       }
     }
     &.fi-text-input--success {
       & .fi-text-input_input {
-        border-color: ${theme.colors.successBase};
+        border: 2px solid ${theme.colors.successBase};
+        padding-left: ${math(`${theme.spacing.insetL} - 1`)};
+        padding-top: 7px;
+        padding-bottom: 7px;
```

Comment:
Not sure if it is really better but might help in future to have the common styles separately from unique styles, like: 
```scss
&.fi-text-input--error,
&.fi-text-input--success {
  & .fi-text-input_input {
    padding-left: ${math(`${theme.spacing.insetL} - 1`)};
    padding-top: 7px;
    padding-bottom: 7px;
  }
}
&.fi-text-input--error {
  & .fi-text-input_input {
    border: 2px solid ${theme.colors.alertBase};
  }
}
&.fi-text-input--success {
  & .fi-text-input_input {
    border: 2px solid ${theme.colors.successBase};
  }
}
```

It is a few more lines, so what do you think about this approach?

Reply:
I gave that some thought too, but since the styles are only needed in two instances that are right next to each other in the document, I thought it cleaner to keep the duplicate code. In the unlikely case there is a third state in the future it would make sense to do things differently, but until then I think there's no value in it in either readability or in saved lines.

Reply:
In my opinion, both work well in this case. Since these states can never overlap, the styles are "kind of not shared". Styles that apply for all states should be shared and seems that they are.